### PR TITLE
Add support for i.MX8QM

### PR DESCRIPTION
**Pull request by paulstelian97**
The i.MX8QM and i.MX8QXP (already supported) platforms are very similar and have only small differences. This pull request contains the changes required to enable support for i.MX8QM.

Firmware PR already merged.

### CODE REVIEW

File: sound/soc/sof/sof-of-dev.c [hunk 1]
```diff
@@ -103,6 +111,7 @@ static int sof_of_remove(struct platform_device *pdev)
 static const struct of_device_id sof_of_ids[] = {
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_IMX8)
 	{ .compatible = "fsl,imx8qxp-dsp", .data = &sof_of_imx8qxp_desc},
+	{ .compatible = "fsl,imx8qm-dsp", .data = &sof_of_imx8qm_desc},
 #endif
```
Comment (dbaluta) [inline sound/soc/sof/sof-of-dev.c:116]: Need to also add documentation bindings as a separate patch, even if that patch won't go through Mark's asoc tree.

Comment (paulstelian97) [reply sound/soc/sof/sof-of-dev.c:116]: Added and pushed, waiting for GH to update PR. You can check the branch directly in the meanwhile.

Comment (dbaluta) [reply sound/soc/sof/sof-of-dev.c:116]: Thanks, looks good to me.

**Comment by dbaluta** [comment]
@paulstelian97 lets make checkpatch.pl a little bit happier

```
CHECK: Please use a blank line after function/struct/union/enum declarations

#29: FILE: sound/soc/sof/sof-of-dev.c:27:

 };

+static struct sof_dev_desc sof_of_imx8qm_desc = {


``

File: sound/soc/sof/sof-of-dev.c [hunk 2]
```diff
@@ -23,6 +24,14 @@ static struct sof_dev_desc sof_of_imx8qxp_desc = {
...
 	.ops = &sof_imx8x_ops,
 };
+
+static struct sof_dev_desc sof_of_imx8qm_desc = {
+	.default_fw_path = "imx/sof",
+	.default_tplg_path = "imx/sof-tplg",
+	.default_fw_filename = "sof-imx8.ri",
+	.nocodec_tplg_filename = "sof-imx8-nocodec.tplg",
+	.ops = &sof_imx8_ops,
+};
```
Comment (plbossart) [inline sound/soc/sof/sof-of-dev.c]: I find this a bit confusing, earlier you mentioned that ' i.MX8QXP which is from i.MX8X", so is this really an IMX8 case?
Are you going to use the same firmware/topology for both?

Comment (dbaluta) [reply sound/soc/sof/sof-of-dev.c]: We will not have the same firmware for i.MX8 (i.MX8QM) / i.MX8X (i.MX8QXP).
Github diff helps with confusion here:

Looking at the code we have:
* for i.MX8 (i.MX8QM) ->  .default_fw_filename = "sof-imx8.ri",
* for i.MX8X (i.MXQXP) -> .default_fw_filename = "sof-imx8x.ri"

Nocodec topology is the same for i.MX8/i.MX8X. Other can be different but it doesn't matter at this point the names
of topology files will be read from device tree.

imx8 naming it is a little bit confusing, but after a while you will get used to it.

Simple tree of i.MX8 boards looks like this:

```
imx8
├── imx8
│   ├── imx8dm (has DSP)
│   └── imx8qm (has DSP)
├── imx8m
│   ├── im8mm
│   ├── im8mn
│   ├── im8mp (has DSP)
│   └── im8mq
└── imx8x
    ├── imx8dxl
    └── imx8qxp (has DSP)
```

**Comment by plbossart** [comment]
@dbaluta I don't mind if you merge, but when you do please be mindful of the two options GitHub provides

a) merge commit 
b) rebase and merge

the case b) help maintain a linear history and makes it easier to squash/prepare patches for upstream. That's the default for both driver and firmware. I only use a) for large merges (e.g. upstream merges or today's SoundWire stuff), so that we can explicitly skip a long list of commits.

These two options are not well documented and I figured I'd let you know.
Thanks!

**Comment by dbaluta** [comment]
@plbossart thanks for pointing this out. Noticed it myself after hitting merge.

 I always do b) for FW changes and somehow github "learned" it. For kernel I think is first time I actually do a merge and github defaulted to a).

Will double check it next time.

**Comment by plbossart** [comment]
no worries, this was a short series.

**Comment by paulstelian97** [comment]
@dbaluta Firmware repo doesn't allow a, that's why the default is b.

**Pull request merged**